dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/data/GetJobData.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * ***************************************************************************
+ *   Copyright 2014-2017 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ***************************************************************************
+ */
+package com.spectralogic.dsbrowser.gui.services.jobService.data
+
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.spectralogic.ds3client.commands.spectrads3.GetActiveJobSpectraS3Request
+import com.spectralogic.ds3client.helpers.Ds3ClientHelpers
+import com.spectralogic.ds3client.helpers.FileObjectGetter
+import com.spectralogic.ds3client.helpers.channelbuilders.PrefixRemoverObjectChannelBuilder
+import com.spectralogic.ds3client.models.bulk.Ds3Object
+import com.spectralogic.dsbrowser.api.services.logging.LoggingService
+import com.spectralogic.dsbrowser.gui.services.jobService.JobTaskElement
+import com.spectralogic.dsbrowser.gui.services.jobService.util.DelimChannelBuilder
+import com.spectralogic.dsbrowser.gui.services.jobService.util.EmptyErrorChannelBuilder
+import com.spectralogic.dsbrowser.gui.util.DateTimeUtils
+import com.spectralogic.dsbrowser.gui.util.ParseJobInterruptionMap
+import javafx.beans.property.SimpleBooleanProperty
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.time.Instant
+
+data class GetJobData(private val list: List<Pair<String, String>>,
+                      private val localPath: Path,
+                      val bucket: String,
+                      private val jte: JobTaskElement) : JobData {
+
+    override var job: Ds3ClientHelpers.Job? = null
+        get() {
+            if (field == null) {
+                field = Ds3ClientHelpers.wrap(jte.client).startReadJob(bucket, buildDs3Objects())
+            }
+            return field!!
+        }
+        set(value) {
+            if (value != null) field = value
+        }
+
+    override fun showCachedJobProperty(): SimpleBooleanProperty = SimpleBooleanProperty(true)
+    override fun loggingService(): LoggingService = jte.loggingService
+    override fun targetPath(): String = localPath.toString()
+    override fun dateTimeUtils(): DateTimeUtils = jte.dateTimeUtils
+    private var startTime = Instant.now()
+    private val last: String? = null
+    override var prefixMap: MutableMap<String, Path> = ImmutableMap.of()
```

Comment:
Was this meant to be Immutable?  If not, why are we returning immutable maps for a mutable type?